client/matchmaking/admin-matchmaking-times.tsx

```diff
@@ -0,0 +1,658 @@
+import {
+  AddMatchmakingTimeRequest,
+  GetFutureMatchmakingTimesResponse,
+  GetMatchmakingTimesResponse,
+  GetPastMatchmakingTimesResponse,
+  MatchmakingTimeJson,
+} from '../../common/matchmaking/matchmaking-times'
+import { apiUrl } from '../../common/urls'
+import { ThunkAction } from '../dispatch-registry'
+import { longTimestamp } from '../i18n/date-formats'
+import { MaterialIcon } from '../icons/material/material-icon'
+import { ElevatedButton, TextButton } from '../material/button'
+import { CheckBox } from '../material/check-box'
+import { TabItem, Tabs } from '../material/tabs'
+import { abortableThunk, RequestHandlingSpec } from '../network/abortable-thunk'
+import { encodeBodyAsParams, fetchJson } from '../network/fetch'
+import { LoadingDotsArea } from '../progress/dots'
+import { useImmerState } from '../react/state-hooks'
+import { useAppDispatch } from '../redux-hooks'
+import { externalShowSnackbar } from '../snackbars/snackbar-controller-registry'
+import { CenteredContentContainer } from '../styles/centered-container'
+import { styledWithAttrs } from '../styles/styled-with-attrs'
+import { bodyLarge, bodyMedium, titleLarge } from '../styles/typography'
+
+function getMatchmakingTimesHistory(
+  type: MatchmakingType,
+  spec: RequestHandlingSpec<GetMatchmakingTimesResponse>,
+): ThunkAction {
+  return abortableThunk(spec, async () => {
+    return await fetchJson<GetMatchmakingTimesResponse>(
+      apiUrl`matchmaking-times/${encodeURIComponent(type)}`,
+      {
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function getMatchmakingTimesFuture(
+  type: MatchmakingType,
+  offset: number,
+  spec: RequestHandlingSpec<GetFutureMatchmakingTimesResponse>,
+): ThunkAction {
+  return abortableThunk(spec, async () => {
+    return await fetchJson<GetFutureMatchmakingTimesResponse>(
+      apiUrl`matchmaking-times/${encodeURIComponent(type)}/future?offset=${offset}`,
+      {
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function getMatchmakingTimesPast(
+  type: MatchmakingType,
+  offset: number,
+  spec: RequestHandlingSpec<GetPastMatchmakingTimesResponse>,
+): ThunkAction {
+  return abortableThunk(spec, async () => {
+    return await fetchJson<GetPastMatchmakingTimesResponse>(
+      apiUrl`matchmaking-times/${encodeURIComponent(type)}/past?offset=${offset}`,
+      {
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function addMatchmakingTime(
+  type: MatchmakingType,
+  startDate: number,
+  enabled: boolean,
+  spec: RequestHandlingSpec<MatchmakingTimeJson>,
+): ThunkAction {
+  return abortableThunk(spec, async dispatch => {
+    const result = await fetchJson<MatchmakingTimeJson>(
+      `/api/1/matchmaking-times/${encodeURIComponent(type)}`,
+      {
+        method: 'post',
+        body: encodeBodyAsParams<AddMatchmakingTimeRequest>({ startDate, enabled }),
+        signal: spec.signal,
+      },
+    )
+
+    externalShowSnackbar('New matchmaking time created')
```

Comment:
You have a place to show this snackbar that is inside the component tree, so please don't use this method. Apply throughout